events_webhook/src/test/java/com/adobe/aio/event/webhook/service/EventVerifierTest.java

```diff
@@ -0,0 +1,172 @@
+  private static final String TEST_DIGI_SIGN_2 = "Xx8uVpZlKIOqAdVBr/6aNrASk6u7i/Bb9kWZttIFOu0Y2JGozZGG7WF9Z6056RdeeBUXLJsV4r8a3ZeEUrOZi3hvhV+Hw7vmK1NIQJVIqdigF9mJ/2gSMGe7K4OPedh+fPNZmbOyNIc6FRmUtTdemNLJeCzM7Zf+niC7Tfsytsz4lW4ebv34TWHxzAA9pZRcJE4a1YYqEYAqn3cHTvCzB/AQ6VdIcP8MsuTGatCk9Vc6dTPOVEcyYkVXTMGgsmzW8RB6mq0m1aqTz3KvnhEYlkspqtxi+jBkTjcYVf1dPa4ofbosmD5rohIef/UwPX5n5ZHM7du86Gf+6S72ee8tbw==";
+  private static final String TEST_INVALID_DIGI_SIGN_1 = "abc22OGm8/6H6bJXSi+/4VztsPN+fPZtHgHrrASuTw7LTUZVpbAZNaXVTzQsFd47PvaI8aQxbl874GFmH0QfAVQaRT93x5O/kQdM1ymG03303QaFY/mjm/Iot3VEwq5xOtM8f5a2mKUce9bgEv28iN7z9H/MbBOSmukPSJh/vMLkFAmMZQwdP4SRK3ckxQg6wWTbeMRxjw8/FLckznCGPZri4c0O7WPr8wnrWcvArlhBpIPJPeifJOyDj/woFQzoeemdrVoBFOieE/j3RoMWzcQeLENaSrqk00MPL2svNQcTLMkmWuICOjYSbnlv/EPFCQS8bQsnVHxGFD1yDeFa7Q==";
+  private static final String TEST_PUB_KEY1_PATH = "/junit/pub-key-1.pem";
+  private static final String TEST_PUB_KEY2_PATH = "/junit/pub-key-2.pem";
+  private static final String TEST_INVALID_PUB_KEY_PATH = "/junit/invalid-key.pem";
+
+  static final int ADOBEIO_CDN_PORT = 9000;
+
+  private EventVerifier underTest;
+
+  private WireMockServer wireMockServer;
+
+  @BeforeEach
+  public void setup() {
+    wireMockServer = new WireMockServer(options().port(ADOBEIO_CDN_PORT));
+    wireMockServer.start();
+    setupEndpointStub();
+    underTest = new EventVerifier("http://localhost:" + ADOBEIO_CDN_PORT);
+  }
+
+  @AfterEach
+  void tearDown() {
+    wireMockServer.stop();
+  }
+
+  private void setupEndpointStub() {
+    configureFor("localhost", ADOBEIO_CDN_PORT);
+    stubFor(get(urlEqualTo(TEST_PUB_KEY1_PATH))
+        .willReturn(aResponse().withBody(getPubKey1())));
+    stubFor(get(urlEqualTo(TEST_PUB_KEY2_PATH))
+        .willReturn(aResponse().withBody(getPubKey2())));
+    stubFor(get(urlEqualTo(TEST_INVALID_PUB_KEY_PATH))
+        .willReturn(aResponse().withBody(getInvalidPubKey())));
+  }
+  @Test
+  public void testVerifyValidSignature() {
+    String message = getTestMessage();
+    Map<String, String> headers = getTestHeadersWithValidSignature();
+    boolean result = underTest.authenticateEvent(message, TEST_CLIENT_ID, headers);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testVerifyInvalidSignature() {
+    String message = getTestMessage();
+    Map<String, String> headers = getTestHeadersWithInvalidSignature();
+    boolean result = underTest.authenticateEvent(message, TEST_CLIENT_ID, headers);
+    assertFalse(result);
+  }
+
+  @Test
+  public void testVerifyInvalidPublicKey() {
+    String message = getTestMessage();
+    Map<String, String> headers = getTestHeadersWithInvalidPubKey();
+    boolean result = underTest.authenticateEvent(message, TEST_CLIENT_ID, headers);
+    assertFalse(result);
+  }
+
+  @Test
+  public void testVerifyInvalidRecipientClient() {
+    String message = getTestMessage();
+    Map<String, String> headers = getTestHeadersWithInvalidPubKey();
```

Comment:
let's use `getTestHeadersWithValidSignature()` instead here
as we only want to test invalid apiKey, right ?

Reply:
done